third_party/move/move-binary-format/src/serializer.rs

```diff
@@ -867,22 +867,94 @@ fn serialize_ability_sets(binary: &mut BinaryData, sets: &[AbilitySet]) -> Resul
 fn serialize_function_attributes(
     binary: &mut BinaryData,
     attributes: &[FunctionAttribute],
+    version: u32,
 ) -> Result<()> {
     write_as_uleb128(binary, attributes.len() as u64, ATTRIBUTE_COUNT_MAX)?;
     for attr in attributes {
-        serialize_function_attribute(binary, attr)?;
+        serialize_function_attribute(binary, attr, version)?;
     }
     Ok(())
 }
 
+fn serialize_function_attribute_if_supported<F>(
```

Comment:
This function could be further generalized to include all function attributes, so that function attribute version checks are all together and not spread out at various callers (we check further up the stack about 2 of the function attributes, the rest are checked in this function's caller).

E.g., this function could take the min_version and bytecode version. It's caller would have min_version for various attributes.